packages/qdrant-loader/src/qdrant_loader/core/state/transitions.py

```
@@ -175,109 +177,186 @@ async def update_document_state(
+                async with session.begin_nested():
+                    record = await _apply_document_state_update(
+                        session, document=document, project_id=project_id
+                    )
+                results.append((document, record, None))
+            except Exception as e:  # noqa: BLE001 - reported to caller, not swallowed
+                results.append((document, None, e))
+
+        try:
+            await session.commit()
+        except (
+            Exception
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)